docs/fidesops/docs/guides/saas_config.md

```diff
@@ -0,0 +1,372 @@
+- `description` Used to add a useful description.
+- `version` Used to track different versions of the SaaS config.
+
+And the following complex fields which we will cover in detail below:
+
+- `connector_params`
+- `client_config`
+- `test_request`
+- `endpoints`
+#### Connector params
+The `connector_params` field is used to describe a list of settings which a user must configure as part of the setup. This section should just include the name of the parameter but not the actual value. These are added as part of the ConnectionConfig [secrets](database_connectors.md#set-the-connectionconfigs-secrets).
+
+```yaml
+connector_params:
+  - name: host
+  - name: username
+  - name: password
+```
+
+#### Client config
+The `client_config` describes the necessary information to be able to create a base HTTP client. Notice that the values for host, username, and password are not defined here, only references in the form of a `connector_param` which Fidesops uses to insert the actual value from the stored secrets.
+
+```yaml
+client_config:
+  protocol: https
+  host:
+    connector_param: host
+  authentication:
+    strategy: basic_authentication
+    configuration:
+      username:
+        connector_param: username
+      password:
+        connector_param: password
+```
+
+The authentication strategies are swappable. In this example we used the `basic_authentication` strategy which uses a `username` and `password` in the configuration. An alternative to this is to use `bearer_authentication` which looks like this:
+```yaml
+authentication:
+  strategy: bearer_authentication
+  configuration:
+    token:
+      connector_param: api_key
+```
+
+#### Test request
+Once the base client is defined we can use a `test_request` to verify our hostname and credentials. This is in the form of an idempotent request (usually a read). The testing approach is the same for any [ConnectionConfig test](database_connectors.md#testing-your-connection).
+```yaml
+test_request:
+  path: /3.0/lists
+```
+#### Endpoints
+This is where we define how we are going to access and update each collection in the corresponding Dataset. The endpoint section contains the following members:
+
+- `name` This name corresponds to a Collection in the corresponding Dataset.
+- `requests` A map of read and update requests for this collection. Each collection can define a way to read and a way to update the data. Each request is made up of:
+    - `path` A static or dynamic resource path. The dynamic portions of the path are enclosed within angle brackets `<dynamic_value>` and are replaced with values from request_params.
+    - `request_params`
+        - `name` Used as the key for query param values, or to map this param to a value placeholder in the path.
+        - `type` Either "query" or "path".
+        - `references` This is the same as reference in the Dataset schema. It is used to define the source of the value for the given request_param.
```

Comment:
Maybe "These are the same as `references` in the.. " or  "This is the same as a `reference` in the"

Reply:
good catch! updating now

Reply:
using `references` since that's the correct key name in the `Dataset` schema